src/stacker/ConcreteStacker.ts

```diff
@@ -54,7 +75,17 @@ export class ConcreteStacker implements Stacker {
   }
 
   async createOrUpdatePRContentsForSingleCommit(commit: Commit): Promise<void> {
-    // TODO: Implement
+    let branch : BranchName = ""; 
+    commit.refNames.forEach(refName => {
+      if(this.isSttackBranch(this.localRefToBranchName(refName))) {
+        branch = this.localRefToBranchName(refName); 
+      }
+    });
+    if(branch === ""){
+      branch = this.createSttackBranch(commit);
+      this.sourceControl.createSttackBranchForCommit(branch, commit); 
+    }
```

Comment:
I know we decided to implement this here on Discord but I didn't realize the code was so Git-specific (specifically the ref <-> branch stuff) 😆. I don't think this belongs in Stacker anymore. It'll be cleaner to move all of that into `GitSourceControl`. We can then just have more abstracted code here:

```suggestion
    const commitBranchPairs = await this.sourceControl.attachSttackBranchesToCommits([commit]);
```

This assumes that `attachSttackBranchesToCommits` will be modified to return something like `Promise<{ commit: Commit, sttackBranch: BranchName }>`